shared/common-adapters/copy-text.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+import * as React from 'react'
+import {Box2} from './box'
+import Button from './button'
+import Text from './text'
+import Icon from './icon'
+import HOCTimers, {type PropsWithTimer} from './hoc-timers'
+import Toast from './toast'
+import {copyToClipboard} from '../util/clipboard'
+import {
+  collapseStyles,
+  type StylesCrossPlatform,
+  globalColors,
+  globalMargins,
+  globalStyles,
+  isMobile,
+  platformStyles,
+  styleSheetCreate,
+} from '../styles'
+
+type Props = PropsWithTimer<{
+  containerStyle?: StylesCrossPlatform,
+  withReveal?: boolean,
+  text: string,
+}>
+
+type State = {
+  showingToast: boolean,
+  revealed: boolean,
+}
+
+class _CopyText extends React.Component<Props, State> {
+  state = {
+    revealed: false,
+    showingToast: false,
+  }
+  _attachmentRef = null
+
+  componendDidMount() {
+    if (!this.props.withReveal) {
+      this.setState({revealed: true})
+    }
+  }
+
+  copy = () => {
+    this.setState({showingToast: true}, () =>
+      this.props.setTimeout(() => this.setState({showingToast: false}), 1500)
+    )
+    copyToClipboard(this.props.text)
+  }
+
+  reveal = () => {
+    this.setState({revealed: true})
+  }
+
+  render() {
+    return (
+      <Box2
+        ref={r => (this._attachmentRef = r)}
+        direction="horizontal"
+        style={collapseStyles([styles.container, this.props.containerStyle])}
+      >
+        <Toast position="top center" attachTo={this._attachmentRef} visible={this.state.showingToast}>
+          {isMobile && <Icon type="iconfont-clipboard" color="white" fontSize={22} />}
+          <Text
+            type={isMobile ? 'BodySmallSemibold' : 'BodySmall'}
+            style={{color: globalColors.white, paddingLeft: 10, paddingRight: 10}}
+          >
+            Copied to clipboard
+          </Text>
+        </Toast>
+        <Text
+          type="Body"
+          selectable={true}
+          style={collapseStyles([styles.text, !this.state.revealed && {width: 'auto'}])}
+        >
+          {this.state.revealed ? this.props.text : '••••••••••••'}
+        </Text>
+        {!this.state.revealed && (
```

Comment:
Should `withReveal` be here too? My assumption, based on its name, is that one with `withReveal: false` wouldn't have a “Reveal” option at all.

Reply:
Added